### PR TITLE
Add control plane APIs support

### PR DESCRIPTION
**Pull request by lxiaopei**

### CODE REVIEW

**Comment by lxiaopei** [comment]
@mengdie-song  @annakhm   Could you help review the change? thanks a lot.

File: api_cluster_control_plane.go [hunk 1]
```diff
@@ -0,0 +1,755 @@
...
+		return localVarReturnValue, localVarHttpResponse, err
+	}
+
+	if localVarHttpResponse.StatusCode >= 300 {
+		newErr := GenericSwaggerError{
+			body:  localVarBody,
+			error: localVarHttpResponse.Status,
+		}
+
+		if localVarHttpResponse.StatusCode == 201 {
```
Comment (mengdie-song) [inline api_cluster_control_plane.go:107]: This line will not be reached.

File: api_cluster_control_plane.go [hunk 2]
```diff
@@ -0,0 +1,755 @@
...
+			err = a.client.decode(&v, localVarBody, localVarHttpResponse.Header.Get("Content-Type"))
+			if err != nil {
+				newErr.error = err.Error()
+				return localVarReturnValue, localVarHttpResponse, newErr
+			}
+			newErr.model = v
+			return localVarReturnValue, localVarHttpResponse, newErr
+		}
+
+		if localVarHttpResponse.StatusCode == 400 {
```
Comment (mengdie-song) [inline api_cluster_control_plane.go:118]: There are some duplicate code for different status codes(400, 403 ...). They can be merged.

File: api_cluster_control_plane.go [hunk 3]
```diff
@@ -0,0 +1,755 @@
...
```
Comment (mengdie-song) [inline api_cluster_control_plane.go:392]: Ditto.

File: api_cluster_control_plane.go [hunk 4]
```diff
@@ -0,0 +1,755 @@
...
+		return localVarReturnValue, localVarHttpResponse, err
+	}
+
+	if localVarHttpResponse.StatusCode >= 300 {
+		newErr := GenericSwaggerError{
+			body:  localVarBody,
+			error: localVarHttpResponse.Status,
+		}
+
+		if localVarHttpResponse.StatusCode == 200 {
```
Comment (mengdie-song) [inline api_cluster_control_plane.go:381]: Ditto.

File: api_client.go [hunk 5]
```diff
@@ -515,6 +516,7 @@ func (c *APIClient) prepareRequest(
 
 	// Generate a new request
 	if body != nil {
+		log.Printf("[DEBUG] the request to nsx-t to register method: %s, url: %s, body: %s", method, url.String(), body)
```
Comment (mengdie-song) [inline api_client.go:519]: I am not sure if this log description is appropriate.

File: api_client.go [hunk 6]
```diff
@@ -560,18 +562,22 @@ func (c *APIClient) prepareRequest(
 		// Basic HTTP Authentication
 		if auth, ok := ctx.Value(ContextBasicAuth).(BasicAuth); ok {
 			localVarRequest.SetBasicAuth(auth.UserName, auth.Password)
+			log.Printf("[DEBUG] basic auth: %s - %s", auth.UserName, auth.Password)
```
Comment (annakhm) [inline api_client.go:565]: We should not reveal sensitive info in logs, same below

**Comment by annakhm** [comment]
Would you like to consider using https://github.com/vmware/vsphere-automation-sdk-go/tree/master/services/nsxt-mp for this?

**Comment by annakhm** [comment]
Would you like to consider using https://github.com/vmware/vsphere-automation-sdk-go/tree/master/services/nsxt-mp for this?

File: api_cluster_control_plane.go (hunk 1 above)
Comment (lxiaopei) [reply api_cluster_control_plane.go:107]: thanks, remove it

File: api_cluster_control_plane.go (hunk 2 above)
Comment (lxiaopei) [reply api_cluster_control_plane.go:118]: thanks, merge the code.

File: api_cluster_control_plane.go (hunk 3 above)
Comment (lxiaopei) [reply api_cluster_control_plane.go:392]: thanks, merge the code.

File: api_cluster_control_plane.go (hunk 4 above)
Comment (lxiaopei) [reply api_cluster_control_plane.go:381]: thanks, merge the code.

File: api_client.go (hunk 5 above)
Comment (lxiaopei) [reply api_client.go:519]: forget to remove this debug message. remove it.

File: api_client.go (hunk 6 above)
Comment (lxiaopei) [reply api_client.go]: remove the debug message.

**Comment by lxiaopei** [comment]
> Would you like to consider using https://github.com/vmware/vsphere-automation-sdk-go/tree/master/services/nsxt-mp for this?

thanks @annakhm  @mengdie-song  for the review.  Addressed the comments. 
After this is merged , I will add the api in https://github.com/vmware/vsphere-automation-sdk-go

**Review by annakhm — APPROVED** [review]

**Review by mengdie-song — COMMENTED** [review]
@lxiaopei  I see there are eight commits in this PR? Could you update this?

**Comment by lxiaopei** [comment]
> @lxiaopei I see there are eight commits in this PR? Could you update this?

merged into 1 commit.

**Review by mengdie-song — APPROVED** [review]

**Pull request merged**